tests/tests/tier0/monitor-node-reconnect/test_monitor_node_reconnect.py

```diff
@@ -0,0 +1,50 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+import os
+import time
+import pytest
+from typing import Dict
+
+from bluechi_test.test import BluechiTest
+from bluechi_test.container import BluechiControllerContainer, BluechiNodeContainer
+from bluechi_test.config import BluechiControllerConfig, BluechiNodeConfig
+
+
+node_name = "node-foo"
+
+
+def exec(ctrl: BluechiControllerContainer, nodes: Dict[str, BluechiNodeContainer]):
+    result, output = ctrl.run_python(os.path.join("python", "is_node_connected.py"))
+    if result != 0:
+        raise Exception(output)
+
+    ctrl.exec_run("systemctl stop bluechi")
+    _, output = ctrl.exec_run('systemctl is-active bluechi')
+    assert output == 'inactive'
+
+    ctrl.exec_run("systemctl start bluechi")
+    ctrl.wait_for_bluechi()
+    # wait a short while for bluechi-agent to reconnect
+    time.sleep(1)
+
+    result, output = ctrl.run_python(os.path.join("python", "is_node_connected.py"))
+    if result != 0:
+        raise Exception(output)
+
+
+@pytest.mark.timeout(15)
+def test_monitor_node_disconnect(
+        bluechi_test: BluechiTest,
+        bluechi_ctrl_default_config: BluechiControllerConfig,
+        bluechi_node_default_config: BluechiNodeConfig):
+
+    node_foo_config = bluechi_node_default_config.deep_copy()
+    node_foo_config.node_name = node_name
+    node_foo_config.heartbeat_interval = "500"
```

Comment:
as we are working in the heartbeat_interval here, can we trigger another test with "50O" or "50A" as value? If you prefer, I can send a PR later.

Reply:
Yes, this would be another test case for #416. 

> If you prefer, I can send a PR later.

That would be great! I was thinking about merging all current invalid option tests into one where a lot of container with all these invalid values are started. What do you think? @dougsland